backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -705,8 +706,25 @@ public InternalFuture<GetUrlForArtifact.Response> getUrlForArtifact(GetUrlForArt
               Collections.singletonList(runId), ModelDBActionEnum.ModelDBServiceActions.UPDATE);
     }
 
-    return permissionCheck.thenCompose(
-        unused -> artifactHandler.getUrlForArtifact(request), executor);
+    return permissionCheck
+        .thenCompose(
+            unused ->
+                artifactHandler.getUrlForArtifact(
+                    ai.verta.common.GetUrlForArtifact.newBuilder()
+                        .setId(request.getId())
+                        .setKey(request.getKey())
+                        .setMethod(request.getMethod())
+                        .setPartNumber(request.getPartNumber())
+                        .setArtifactType(request.getArtifactType())
+                        .build()),
+            executor)
+        .thenApply(
+            response ->
+                GetUrlForArtifact.Response.newBuilder()
+                    .setUrl(response.getUrl())
+                    .setMultipartUploadOk(response.getMultipartUploadOk())
+                    .build(),
+            executor);
```

Comment:
this also looks like a functional change.

Reply:
this change is part of common module and we have added protos changes here: https://github.com/VertaAI/protos-all/pull/336 to make Artifact store code in common module and then we can use it in java like above

Reply:
tests needed